src/Plugins/ComplexCore/src/ComplexCore/Filters/CropImageGeometry.cpp

```diff
@@ -0,0 +1,492 @@
+
+  params.insert(std::make_unique<BoolParameter>(k_RenumberFeatures_Key, "Renumber Features", "Specifies if the feature IDs should be renumbered", false));
+  params.insert(std::make_unique<ArraySelectionParameter>(k_FeatureIds_Key, "Feature IDs", "DataPath to the target Feature IDs array", DataPath()));
+
+  params.insert(std::make_unique<StringParameter>(k_NewFeaturesName_Key, "New Cell Features Group Name", "Name of the new DataGroup containing updated updated Voxel Arrays", "Cell Features"));
+  return params;
+}
+
+IFilter::UniquePointer CropImageGeometry::clone() const
+{
+  return std::make_unique<CropImageGeometry>();
+}
+
+IFilter::PreflightResult CropImageGeometry::preflightImpl(const DataStructure& data, const Arguments& args, const MessageHandler& messageHandler) const
+{
+  auto srcImagePath = args.value<DataPath>(k_ImageGeom_Key);
+  auto destImagePath = args.value<DataPath>(k_NewImageGeom_Key);
+  auto featureIdsArrayPath = args.value<DataPath>(k_FeatureIds_Key);
+  auto xMax = args.value<int32>(k_MinX_Key);
+  auto xMin = args.value<int32>(k_MaxX_Key);
+  auto yMax = args.value<int32>(k_MaxY_Key);
+  auto yMin = args.value<int32>(k_MinY_Key);
+  auto zMax = args.value<int32>(k_MaxZ_Key);
+  auto zMin = args.value<int32>(k_MinZ_Key);
+  auto shouldUpdateOrigin = args.value<bool>(k_UpdateOrigin_Key);
+  auto shouldRenumberFeatures = args.value<bool>(k_RenumberFeatures_Key);
+  auto voxelArrayPaths = args.value<std::vector<DataPath>>(k_VoxelArrays_Key);
+  auto newCellFeaturesName = args.value<std::string>(k_NewFeaturesName_Key);
+
+  OutputActions actions;
+
+  if(xMax < xMin)
+  {
+    std::string ss = fmt::format("X Max (%1) less than X Min (%2)", xMax, xMin);
+    return {MakeErrorResult<OutputActions>(-5550, ss)};
+  }
+  if(yMax < yMin)
+  {
+    std::string ss = fmt::format("Y Max ({}) less than Y Min ({})", yMax, yMin);
+    return {MakeErrorResult<OutputActions>(-5550, ss)};
+  }
+  if(zMax < zMin)
+  {
+    std::string ss = fmt::format("Z Max ({}) less than Z Min ({})", zMax, zMin);
+    return {MakeErrorResult<OutputActions>(-5550, ss)};
+  }
+  if(xMin < 0)
+  {
+    std::string ss = fmt::format("X Min ({}) less than 0", xMin);
+    return {MakeErrorResult<OutputActions>(-5550, ss)};
+  }
+  if(yMin < 0)
+  {
+    std::string ss = fmt::format("Y Min ({}) less than 0", yMin);
+    return {MakeErrorResult<OutputActions>(-5550, ss)};
+  }
+  if(zMin < 0)
+  {
+    std::string ss = fmt::format("Z Min ({}) less than 0", zMin);
+    return {MakeErrorResult<OutputActions>(-5550, ss)};
+  }
+
+  // Validate the incoming DataContainer, Geometry, and AttributeMatrix ; bail if any do not exist since we plan on using them later on in the dataCheck
+  auto oldDimensions = getCurrentVolumeDataContainerDimensions(data, srcImagePath);
+  auto oldResolution = getCurrentVolumeDataContainerResolutions(data, srcImagePath);
```

Comment:
The comment indicates that these are required, but the functions don't have a way to report errors.